packages/react-native-sdk/docusaurus/docs/reactnative/02-tutorials/02-audio-room.mdx

```diff
@@ -1,9 +1,1111 @@
+
+<TokenSnippet sampleApp='audio-rooms' displayStyle='credentials' />
+
+With valid credentials in place, we can create or join an audio room call.
+
+### Step 3 - Add a modal view to create a new audio room call
+
+Let us create a Modal that allows a user to create a new audio room call. Create a new file `src/CreateRoomModal.tsx` and replace it with this code:
+
+```tsx title="src/CreateRoomModal.tsx"
+import {
+  Call,
+  GetOrCreateCallRequest,
+  useStreamVideoClient,
+} from '@stream-io/video-react-native-sdk';
+import React, { useState } from 'react';
+import {
+  Modal,
+  Pressable,
+  StyleSheet,
+  View,
+  Button,
+  TextInput,
+} from 'react-native';
+import { user } from "./constants"
+
+// generate a random id
+const generateRoomId = () => Math.random().toString(36).substring(2, 12);
+
+// create payload for creating a new call
+const generateRoomPayload = ({
+  title,
+  description,
+}: {
+  title?: string;
+  description?: string;
+}): GetOrCreateCallRequest => {
+  return {
+    data: {
+      members: [{ user_id: user.id, role: 'admin' }],
+      custom: {
+        title: title || `${user.name}'s Room`,
+        description: description || `Room created by ${user.name}.`,
+        hosts: [user],
+      },
+    },
+  };
+};
+
+type Props = {
+  modalVisible: boolean;
+  setCall: (call: Call) => void;
+  onClose: () => void;
+};
+
+export default function CreateRoomModal(props: Props) {
+  const client = useStreamVideoClient();
+
+  const [title, setTitle] = useState<string>();
+  const [description, setDescription] = useState<string>();
+
+  const createRoom = () => {
+    if (!client) {
+      return;
+    }
+    const call = client.call('audio_room', generateRoomId());
+    call.getOrCreate(generateRoomPayload({ user, title, description }));
+    props.setCall(call);
+  };
+
+  return (
+    <Modal
+      animationType="fade"
+      transparent={true}
+      visible={props.modalVisible}
+      onRequestClose={props.onClose}
+    >
+      <Pressable style={styles.centeredView} onPress={props.onClose}>
+        <View style={styles.modalView}>
+          <TextInput
+            placeholder={'Type the title of the room'}
+            value={title}
+            style={styles.textInputTitle}
+            autoCorrect={false}
+            placeholderTextColor="#808080"
+            underlineColorAndroid="transparent"
+            onChangeText={setTitle}
+          />
+          <TextInput
+            placeholder={'Type the description of the room'}
+            value={description}
+            multiline={true}
+            autoCorrect={false}
+            placeholderTextColor="#808080"
+            underlineColorAndroid="transparent"
+            style={styles.textInputDescription}
+            onChangeText={setDescription}
+          />
+          <Button onPress={createRoom} title="Create" />
+        </View>
+      </Pressable>
+    </Modal>
+  );
+}
+
+const styles = StyleSheet.create({
```

Comment:
Not sure if we should add all this styling in a tutorial. It should be very minimal, in my opinion. Users are free to add there own if they want to. This also increases the length of the tutorial unnecessarily.

Reply:
React uses their css. If we dont add any styling what is seen on the screenshot is not what you get from the tutorial. 

Basically tutorial needs some styling, if we were web we could just give out a CSS file. We are react-native, what to do :(